### ISSUE TITLE
libmtp error:  Error getting friendlyname.

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#60384]
## Issue description

I tried to copy files from my Android smartphone to my notebook using Pantheon Files.

For some files (all videos) i get this error:

`libmtp error:  Error getting friendlyname.`

It is reproducible with this files.

syslog:

```
Apr 28 20:08:23 X230 org.gtk.vfs.Daemon[1924]: mkdir failed on directory /var/cache/samba: Permission denied
Apr 28 20:08:23 X230 org.gtk.vfs.Daemon[1924]: mkdir failed on directory /var/cache/samba: Permission denied
Apr 28 20:08:23 X230 .io.elementary.[12341]: FileUtils.vala:311: Invalid mtp path
Apr 28 20:08:24 X230 .io.elementary.[12341]: FileUtils.vala:311: Invalid mtp path
Apr 28 20:08:26 X230 org.gtk.vfs.Daemon[1924]: mkdir failed on directory /var/cache/samba: Permission denied
Apr 28 20:08:26 X230 org.gtk.vfs.Daemon[1924]: mkdir failed on directory /var/cache/samba: Permission denied
Apr 28 20:08:29 X230 .io.elementary.[12341]: Sidebar.vala:932: Error querying smb://nas.lan/media/ filesystem info: No route to host
```

I noticed that the files seem to be broken. The video player on the smartphone stops playing in the middle...

@worldofpeace any idea?

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.36, NixOS, 19.03.172361.cf3e277dd0b (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(davidak): `""`
 - channels(root): `"nixos-19.03.172361.cf3e277dd0b, nixos-hardware, nixos-unstable-19.09pre177249.dfd8f84aef1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by worldofpeace** [comment]
> I noticed that the files seem to be broken. The video player on the smartphone stops playing in the middle...

That is the files which are videos that you tried to copy, correct?
If so that sounds like a pretty dangerous issue.

**Comment by davidak** [comment]
>That is the files which are videos that you tried to copy, correct?

 yes

>If so that sounds like a pretty dangerous issue.

why dangerous? i think since they are broken, it's not a big deal that i can't copy them. but i think it should be possible in general to copy even broken files

**Comment by worldofpeace** [comment]
> > If so that sounds like a pretty dangerous issue.
> 
> why dangerous? i think since they are broken, it's not a big deal that i can't copy them. but i think it should be possible in general to copy even broken files

Oh now I understand. I was under the assumption that because you tried to copy them and it failed that it resulted in some sort of corruption.
Can confirm that video files of the same file format that aren't corrupted can by copied?

Also, to find out if it's actually an issue in `libmtp` or really with like `Pantheon Files` and `gvfs` can you try to do the same operation but only with `libmtp` cli?

##### I think the https://wiki.archlinux.org/index.php/Media_Transfer_Protocol section `libmtp` should be helpful

**Comment by davidak** [comment]
>Can confirm that video files of the same file format that aren't corrupted can by copied?

yes. i copied some hundred files and only 5 or so had this problem

> can you try to do the same operation but only with libmtp cli?

i already have deleted all files from my smartphone. so, no

**Comment by alexpear** [comment]
Still important to me. Seen with a .jpg file in a similar file transfer situation.
